Source/RapyutaSimulationPlugins/Private/Core/RRROS2GameMode.cpp

```diff
@@ -59,6 +60,6 @@ void ARRROS2GameMode::InitROS2()
     ClockPublisher->InitializeWithROS2(ROS2Node);
 
     // Simulation state
-    SimulationState = currentWorld->SpawnActor<ASimulationState>();
+    SimulationState = currentWorld->SpawnActor<ASimulationState>(SimulationStateClass);
```

Comment:
The spawned actor will be of the type indicated by `SimulationStateClass` (similarly as in `NewObject`).
This would facilitate the creation of instances of an `ASimulationState`'s child class like `ARRInternalSimulationState`, which is required for the custom handling of the internal service like `SpawnWorld`.